### PR TITLE
add support for edge functions

### PR DESCRIPTION
**Pull request by hong-yi**
- add support for oidc role to update edge lambdas and cloudfront distributions if required
- fix typo

### CODE REVIEW

**Review by tymnicholas — APPROVED** [review]

File: data.tf [hunk 1]
```diff
@@ -32,3 +32,37 @@ data "aws_iam_policy_document" "update_lambda" {
...
+
+  dynamic "statement" {
+    for_each = var.cf_distribution_id != null ? [] : [1]
+    content {
+      sid = "AllowCloudFrontUpdateDistributionAccess"
+
+      actions = [
+        "cloudfront:UpdateDistribution",
+        "cloudfront:GetDistribution",
+        "cloudfront:GetDistributionConfig"
```
Comment (niroz89) [inline data.tf:59]:
```suggestion
        "cloudfront:GetDistributionConfig",
```

File: data.tf [hunk 2]
```diff
@@ -32,3 +32,37 @@ data "aws_iam_policy_document" "update_lambda" {
...
+      ]
+
+      resources = [
+        "arn:aws:cloudfront::${data.aws_caller_identity.current.account_id}:distribution/${var.cf_distribution_id}"
```
Comment (niroz89) [inline data.tf:63]:
```suggestion
        "arn:aws:cloudfront::${data.aws_caller_identity.current.account_id}:distribution/${var.cf_distribution_id}",
```

File: data.tf [hunk 3]
```diff
@@ -32,3 +32,37 @@ data "aws_iam_policy_document" "update_lambda" {
...
   }
 }
+
+data "aws_iam_policy_document" "update_lambda_edge" {
+  statement {
+    sid = "EnableCFReplication"
+
+    actions = [
+      "lambda:GetFunction",
+      "lambda:EnableReplication"
```
Comment (niroz89) [inline data.tf:42]:
```suggestion
      "lambda:EnableReplication",
```

File: github_action.tf [hunk 4]
```diff
@@ -5,7 +5,7 @@ module "oidc_provider" {
   count = var.create_github_actions_oidc_provider ? 1 : 0
 }
 
-module "lamda_gha" {
+module "lambda_gha" {
```
Comment (niroz89) [inline github_action.tf:8]: 👍🏼

Comment (niroz89) [reply github_action.tf:8]: let's add a `moved` block.

**Review by wayne-root — APPROVED** [review]

**Review by niroz89 — COMMENTED** [review]
Looks good, just cosmetics, can help to generate `terraform-docs` as well.

**Pull request merged**